ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/RemoteFileServiceImpl.java

```diff
@@ -453,8 +467,17 @@ public void getById(String fileId,
 
                 // BAM/CRAM File
                 URL resURL = new URL(resURL() + "/file/archive/" + reqFile.getFileId()); // Just specify file ID
-                SeekableStream cIn = (new EgaSeekableCachedResStream(resURL, null, null, reqFile.getFileSize())).setExtension(extension); // Deals with coordinates
-                //bIn = new SeekableBufferedStream(cIn);
+
+                // HTSJDK works out if the stream is a BAM or a CRAM file from the extension on the URI but RES URIs do
+                // not have extensions, so override the name returned by getSource
+                final String finalExtension = extension;
+                SeekableStream cIn = new SimpleSeekableStream(resURL, client, resDataChunkSize, reqFile.getFileSize()) {
+                    @Override
```

Comment:
Can this override be part of the constructor? I haven't tried it, so I don't know if Java permits it, but maybe something like this:

```java
SeekableStream cIn = new SimpleSeekableStream(resURL, client, resDataChunkSize, reqFile.getFileSize(), finalExtension);
```
And in the SimpleSeekableStream class:
```java
public String getSource() {
  if (finalExtension is not blank) {
    return super.getSource() + "." + finalExtension;
  } else {
    return super.getSource();
  }
}
```
Or is this uglier?